File: project/hyeok-onboarding/src/main/java/com/INNER_CIRCLE_ICD4/innerCircle/survey/request/domain/Response.java
package com.INNER_CIRCLE_ICD4.innerCircle.survey.request.domain;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Response {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id")
    private Survey survey;

    @OneToMany(mappedBy = "response", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Answer> answers = new ArrayList<>();

    // === 연관 관계 메서드 ===
    public void addAnswer(Answer answer) {
        answers.add(answer);
        answer.setResponse(this);
    }

    // === Getters/Setters ===
    public Long getId() {
        return id;
    }

    public Survey getSurvey() {
        return survey;
    }

    public void setSurvey(Survey survey) {
        this.survey = survey;
    }

    public List<Answer> getAnswers() {
        return answers;
    }
}

File: project/jinho-onboarding/src/main/kotlin/fc/innercircle/jinhoonboarding/survey/domain/Response.kt
package fc.innercircle.jinhoonboarding.survey.domain

import jakarta.persistence.*

@Entity
class Response (
    val surveyId: Long,
    val questionId: Long,
    val question: String,

    @OneToMany(fetch = FetchType.LAZY)
    @JoinColumn(name = "response_id")
    val answer: List<Answer>,
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
}